src/object.d

```diff
@@ -2922,21 +2918,53 @@ unittest
     assert(postblitRecurseOrder == order);
 }
 
+// destroy
 /********
-Destroys the given object and sets it back to its initial state. It's used to
-_destroy an object, calling its destructor or finalizer so it no longer
-references any other objects. It does $(I not) initiate a GC cycle or free
-any GC memory.
+Destroys an object by means of calling its destructor, if present. For
+types `T` that define a destructor, `destroy` then also fills the
+object with `T.init` to ensure the object's destructor can be called
+again (e.g. when the object goes out of scope). Unlike $(LREF
+__delete), `destroy` does not release memory or other resources by
+itself (though it is possible that destructors it calls do so).
+
+Note: A `struct` type may have a destructor even if it does not
+directly define `~this()`, if at least one field has a destructor
+(transitively). A `class` type may have a destructor even if it does
+not directly define `~this()`, if at least one field has a destructor
+(transitively), or if the destructor is inherited.
+
+Detailed behavior of `destroy(obj)` depends on the type `T` as
+follows:
+
+$(UL $(LI If `T` is a primitive type, a `struct` types that does not
+have a destructor, or a `class` type that does not have a destructor,
+the call to has no effect.)
```

Comment:
"call to has no effect" Maybe missing "destroy" in there?

Note, this is technically incorrect for a class type with no destructor, as destroying a class instance that has no destructor is still going to call `rt_finalize`, which at least sets the vtable to null.